src/plugins/embeddable/public/embeddable_panel/panel_header/use_embeddable_panel_title.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import classNames from 'classnames';
+import React, { useMemo } from 'react';
+import { EuiIcon, EuiLink, EuiToolTip } from '@elastic/eui';
+
+import { IEmbeddable, ViewMode } from '../../lib';
+import { CustomizePanelAction } from '../panel_actions';
+import { useSelectFromEmbeddableInput } from '../use_select_from_embeddable';
+import { getEditTitleAriaLabel, placeholderTitle } from '../embeddable_panel_strings';
+
+export const useEmbeddablePanelTitle = (
```

Comment:
nit 
Why is this a hook vs just a component? It returns an ReactNode or null. I think making it a normal component vs a hook would make more sense.

Reply:
Good call, will swap this over!